src/main/java/org/openrewrite/java/spring/security5/UseNewRequestMatchers.java

```diff
@@ -57,46 +59,26 @@ public String getDescription() {
-                new UsesMethod<>(REGEX_MATCHERS)), new JavaIsoVisitor<ExecutionContext>() {
-
-            @Override
-            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
-                J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
-                if (ANT_MATCHERS.matches(mi) || MVC_MATCHERS.matches(mi) || REGEX_MATCHERS.matches(mi)) {
-                    mi = maybeChangeMethodInvocation(mi);
-                }
-                return mi;
-            }
-        });
-
-    }
-
-    private J.MethodInvocation maybeChangeMethodInvocation(J.MethodInvocation mi) {
-        return findRequestMatchersMethodWithMatchingParameterTypes(mi)
-                .map(requestMatchersMethod -> mi
-                        .withMethodType(requestMatchersMethod)
-                        .withName(mi.getName().withSimpleName("requestMatchers")))
-                .orElse(mi);
-    }
-
-    private Optional<JavaType.Method> findRequestMatchersMethodWithMatchingParameterTypes(J.MethodInvocation mi) {
-        JavaType.Method methodType = mi.getMethodType();
-        if (methodType == null) {
-            return Optional.empty();
-        }
-        List<JavaType> parameterTypes = methodType.getParameterTypes();
-        List<JavaType.Method> methods;
-        boolean isOverride = TypeUtils.isOverride(mi.getMethodType());
-        if (isOverride) {
-            methods = requireNonNull(methodType.getDeclaringType().getSupertype(), "superType").getMethods();
-        } else {
-            methods = methodType.getDeclaringType().getMethods();
-        }
-        return methods.stream()
-                .filter(m -> m.getName().equals("requestMatchers"))
-                .filter(m -> m.getParameterTypes().equals(parameterTypes))
-                .findFirst();
+                        new UsesMethod<>(ANT_MATCHERS),
+                        new UsesMethod<>(MVC_MATCHERS),
+                        new UsesMethod<>(REGEX_MATCHERS)),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                        J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
+                        if ((ANT_MATCHERS.matches(mi) || MVC_MATCHERS.matches(mi) || REGEX_MATCHERS.matches(mi))
+                                && mi.getSelect() != null) {
+                            String parametersTemplate = mi.getArguments().stream().map(arg -> "#{any()}").collect(joining(", "));
+                            List<Expression> parameters = new ArrayList<>();
+                            parameters.add(mi.getSelect());
+                            parameters.addAll(mi.getArguments());
+                            JavaTemplate template = JavaTemplate.builder(String.format("#{any()}\n.requestMatchers(%s)", parametersTemplate))
```

Comment:
Not proud of the `\n` snuck in here either, but JavaTemplate strips all whitespace (and comments!) from the statements following the replaced method invocation. Would very much appreciate a proper fix, but not sure how JavaTemplate breaks that suffix to begin with. 🤔 

Could be something to pick up later if no one with more expertise is available on short notice.